cmd/mkroot/main.go

```diff
@@ -0,0 +1,148 @@
+package main
+
+import (
+	"crypto/rand"
+	"crypto/sha1"
+	"crypto/x509"
+	"crypto/x509/pkix"
+	"encoding/json"
+	"encoding/pem"
+	"flag"
+	"fmt"
+	"io/ioutil"
+	"math/big"
+	"time"
+
+	"github.com/cloudflare/cfssl/crypto/pkcs11key"
+	"github.com/cloudflare/cfssl/log"
+)
+
+var configFile = flag.String("config", "", "JSON file for subject and validity")
+var module = flag.String("pkcs11-module", "", "PKCS#11 module")
+var pin = flag.String("pkcs11-pin", "", "PKCS#11 password")
+var token = flag.String("pkcs11-token", "", "PKCS#11 token name")
+var label = flag.String("pkcs11-label", "", "PKCS#11 key label")
+
+type Config struct {
+	Name struct {
+		C  string
+		O  string
+		OU string
+		CN string
+	}
+	NotBefore time.Time
+	NotAfter  time.Time
+}
+
+func main() {
+	// Validate input
+	// All flags are required
+	flag.Parse()
+	missing := false
+	switch {
+	case len(*configFile) == 0:
```

Comment:
In future you can simplify this case with `flag.Visitall(fn func(*Flag))`, which visits even unset flags.